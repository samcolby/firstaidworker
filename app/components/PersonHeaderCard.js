import React, { PureComponent } from "react";
import PropTypes from "prop-types";

import { StyleSheet, Text, View } from "react-native";
import { Avatar, Card } from "react-native-elements";

/**
 * React Pure Component to display
 * a header card for a person.
 *
 * This can be used along the top of a screen
 * that shows relevant data for the passed in person.
 *
 * @class PersonHeaderCard
 * @extends {PureComponent}
 */
class PersonHeaderCard extends PureComponent {
  static propTypes = {
    person: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);
  }

  render() {
    const { name, job_title, avatar_uri } = this.props.person;

    return (
      <Card containerStyle={styles.containerStyle}>
        <View style={styles.headerContainer}>
          <Avatar
            activeOpacity={0.7}
            rounded
            size="large"
            source={{ uri: avatar_uri }}
            title={name[0]}
          />
          <View style={styles.headerTextContainer}>
            <Text style={styles.headerTextTitle}>{name}</Text>
            <Text style={styles.headerTextSubtitle}>{job_title}</Text>
          </View>
        </View>
      </Card>
    );
  }
}

const styles = StyleSheet.create({
  containerStyle: {
    borderWidth: 0,
    padding: 0,
    paddingVertical: 15,
    margin: 0
  },
  headerContainer: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
    paddingLeft: 20
  },
  headerTextContainer: {
    fontSize: 20,
    marginLeft: 20,
    flex: 1,
    flexDirection: "column"
  },
  headerTextTitle: { fontSize: 20 },
  headerTextSubtitle: {
    fontSize: 16,
    color: "grey"
  }
});

export default PersonHeaderCard;
