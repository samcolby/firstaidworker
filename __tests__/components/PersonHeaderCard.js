import React from "react";
import { shallow } from "enzyme";

import { Text } from "react-native";
import { Avatar } from "react-native-elements";

import PersonHeaderCard from "../../app/components/PersonHeaderCard";

const testData = {
  id: "c4ab9532-198e-4891-8719-8167005e9fa9",
  name: "Maryellen Baxter",
  job_title: "Flouncer",
  isActive: true,
  picture: "https://randomuser.me/api/portraits/women/12.jpg",
  company: "PANZENT",
  email: "maryellenbaxter@panzent.com",
  phone: "+1 (848) 545-3302",
  latitude: 51.2000548,
  longitude: -4.1474755
};

describe("<PersonHeaderCard /> component", () => {
  it("renders correctly", () => {
    const component = shallow(<PersonHeaderCard person={testData} />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("displays the person's picture", () => {
    const component = shallow(<PersonHeaderCard person={testData} />);

    const avatarComponent = component.find(Avatar);

    expect(avatarComponent.prop("source").uri).toEqual(testData.picture);
  });

  it("displays the person's company", () => {
    const component = shallow(<PersonHeaderCard person={testData} />);

    // const textComponent = component.find(Text).at(0);

    // expect(textComponent.props().children).toEqual(testData.company);

    expect(
      component.containsMatchingElement(<Text>{testData.company}</Text>)
    ).toBe(true);
  });

  it("displays the job title", () => {
    const component = shallow(<PersonHeaderCard person={testData} />);

    expect(
      component.containsMatchingElement(<Text>{testData.job_title}</Text>)
    ).toBe(true);
  });
});
