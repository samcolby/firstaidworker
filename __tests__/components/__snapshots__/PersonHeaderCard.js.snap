// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonHeaderCard /> component renders correctly 1`] = `
<Themed.Card
  containerStyle={
    Object {
      "borderWidth": 0,
      "margin": 0,
      "padding": 0,
      "paddingVertical": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingLeft": 20,
      }
    }
  >
    <Themed.Avatar
      activeOpacity={0.7}
      rounded={true}
      size="large"
      source={
        Object {
          "uri": "https://randomuser.me/api/portraits/women/12.jpg",
        }
      }
      title="M"
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "fontSize": 20,
          "marginLeft": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        PANZENT
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 16,
          }
        }
      >
        Flouncer
      </Text>
    </View>
  </View>
</Themed.Card>
`;
